### PR TITLE
[main.py] Part-1/2, Do not pass args command into functions

### PR DESCRIPTION
**Pull request by ntadimeti**
Pass explicit arguments into functions instead of passing `args`. All except infer_worker, eval_worker and train_worker are changed so that they accept explicit input arguments. 

The `*_workers` are not changed because the list of arguments they need is very big and we would need a data structure (may be a dict) to zip those arguments together. This will be addressed in another PR.

### CODE REVIEW

**Comment by tijyojwad** [comment]
These are some changes for #48

File: main.py [hunk 1]
```diff
@@ -272,7 +298,10 @@ def infer_worker(gpu, ngpu_per_node, args, res_queue=None):
         dist.init_process_group(backend=args.dist_backend, init_method=args.dist_url,
                                 world_size=args.world_size, rank=args.rank)
 
-    model = build_model(args)
+    model = build_model(args.model, args.rank, args.interval_size, args.afunc,
```
Comment (tijyojwad) [inline main.py]: for these long argument functions, let's use named arguments to make the usage clearer (and avoid potential bugs).

e.g. `build_model(model=args.modek, rank=args.rank, interval_size=args.interval_size,...)`. Same for the `writer` function below as well

Comment (ntadimeti) [reply main.py]: we call the writer function using mp.Process. Not sure if I can send in args in the above format ... changed build_model.

Comment (tijyojwad) [reply main.py]: https://stackoverflow.com/questions/11546858/python-multiprocessing-keyword-arguments I believe describes how to pass keyword args to multi process calls. Can you check if this works in your scenario?

Comment (ntadimeti) [reply main.py]: changed it, can you check now ?

Comment (ntadimeti) [reply main.py]: would also appreciate if you can test it quickly to make sure nothing unexpected comes along the way

**Pull request merged**